python/teams_ai/turn_state/turn_state_entry.py

```diff
@@ -3,58 +3,66 @@
 Licensed under the MIT License.
 """
 
-from typing import TypeVar, Generic, TypedDict, Union
+from typing import TypeVar, Generic, Optional
 
-T = TypeVar("T", TypedDict)
+ValueT = TypeVar("ValueT")
 
 
-class TurnStateEntry(Generic[T]):
+class TurnStateEntry(Generic[ValueT]):
     "accessor class for managing an individual state scope"
 
-    __value: T
-    __storage_key: Union[str, None]
-    __deleted = False
-    __hash: str
+    _value: ValueT
+    _storage_key: Optional[str]
+    _hash: str
+    _deleted = False
 
-    def __init__(self, value: T, storage_key=None) -> None:
-        self.__value = value
-        self.__storage_key = storage_key
-        self.__hash = value.__str__()
+    def __init__(self,
+                 value: Optional[ValueT] = None,
+                 storage_key: Optional[str] = None) -> None:
+        self._value = value
+        self._storage_key = storage_key
+        self._hash = str(self._value)
 
     @property
     def changed(self) -> bool:
```

Comment:
The name in JS SDK is `hasChanged`, and DOTNET SDK also names it as `HasChanged` by following language specific naming rule. Should we name the property to `has_changed`?